container/container_contract.go

```diff
@@ -774,9 +800,33 @@ func isStorageNode(ctx storage.Context, key interop.PublicKey) bool {
 	return false
 }
 
-func keysToDelete(ctx storage.Context, epoch int) [][]byte {
-	results := [][]byte{}
+func updateEstimations(ctx storage.Context, epoch int, cid []byte, pub interop.PublicKey, isUpdate bool) {
+	h := crypto.Ripemd160(pub)
+	estKey := append([]byte(singleEstimatePrefix), cid...)
+	estKey = append(estKey, h...)
+
+	var newEpochs []int
+	rawList := storage.Get(ctx, estKey).([]byte)
+
+	if rawList != nil {
+		epochs := std.Deserialize(rawList).([]int)
```

Comment:
Storing slice is efficient, but won't be there any issues if multiple estimations with new epoch value are going to be in the same block? 

It shouldn't though, all tx will have enough GAS to invoke `storage.Delete` but only first tx in the block will do it, and others txs will be cheaper. But maybe I am missing something.

Reply:
My initial thought was that because an estimate is produced by a certain node for a specific container, having multiple tx in one block is highly unlikely, given that epoch is much bigger than block time.
I wanted to get rid of `storage.Find` to reduce GAS cost (this will be amplified by iteration in `newEpoch`), but may be it doesn't worth the effort.